src/DynamoAnalyzer/Analyzer/BinaryAnalyzer.cs

```diff
@@ -0,0 +1,109 @@
+using DynamoAnalyzer.Helper;
+using DynamoAnalyzer.Models;
+using DynamoAnalyzer.Models.UpgradeAssistant;
+
+namespace DynamoAnalyzer.Analyzer
+{
+    /// <summary>
+    /// Implementation of the binary analyzer using the analyzebinaries of the upgrade-assistant tool
+    /// </summary>
+    public class BinaryAnalyzer : IAnalyze
+    {
+        private readonly FileInfo _file;
+        private readonly AnalyzedPackage _package;
+        private readonly string workspace;
+
+        /// <summary>
+        /// Name of the file being analyzed
+        /// </summary>
+        public string Name
+        {
+            get
+            {
+                return _file?.Name;
+            }
+        }
+
+        /// <summary>
+        /// The path of the DLL currently being analyzed
+        /// </summary>
+        public string Path
+        {
+            get
+            {
+                if (_file != null)
+                {
+                    string workdir = System.IO.Path.Combine(AnalyzeEnvironment.GetWorkspace().FullName, System.IO.Path.GetFileNameWithoutExtension(_package.ArchiveName));
+                    return _file.FullName.Replace(workdir + "\\", "").Replace(_file.Name, "");
+                }
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Creates a new BinaryAnalyzer instance
+        /// </summary>
+        /// <param name="file"></param>
+        /// <param name="package"></param>
+        /// <param name="workspace"></param>
+        /// <exception cref="InvalidOperationException"></exception>
+        public BinaryAnalyzer(FileInfo file, AnalyzedPackage package, string workspace)
+        {
+            _file = file;
+            _package = package;
+            if (string.IsNullOrEmpty(workspace))
+            {
+                throw new InvalidOperationException($"{nameof(workspace)} can't be null");
+            }
+            this.workspace = workspace;
+        }
+
+        /// <summary>
+        /// Start the upgrade-assistant tool
+        /// </summary>
+        /// <returns></returns>
+        public async Task<AnalyzedPackage> Process()
+        {
+            AnalyzedPackage dllAnalysis = _package.Copy();
+            dllAnalysis.ArtifactName = Name;
+            dllAnalysis.ArtifactPath = Path;
+
+            try
+            {
+                await ProcessHelper.StartAnalyzeProcess(workspace, GetArgs());
+                FileInfo sarifFile = new FileInfo(System.IO.Path.Combine(workspace, ConfigHelper.GetConfiguration()["SarifFileName"]));
+                Sarif result = await SarifHelper.ReadSarif(sarifFile);
+                Result[] res = result.Runs.SelectMany(f => f.Results).ToArray();
+                dllAnalysis.RequirePort = res.Any();
+
+                List<string> f_res = new List<string>(res.SelectMany(f => f.Locations.Select(g => g.PhysicalLocation.ArtifactLocation.URI.Replace("file:///", "").Replace(workspace.Replace("\\", "/"), ""))).Distinct());
```

Comment:
This is an example of a succesful AnalysisReport.sarif (result of Microsoft's upgrade-assistant tool):
```json
{
    "$schema": "https://schemastore.azurewebsites.net/schemas/json/sarif-2.1.0-rtm.5.json",
    "version": "2.1.0",
    "runs": [{
            "tool": "ignored for explanation purposes"
            },
            "results": [{
                    "ruleId": "UA9011",
                    "level": "note",
                    "message": {
                        "text": "Member System.Drawing.Drawing2D.GraphicsPath.PathTypes.get : [net7.0 Availability: Available via package System.Drawing.Common] [Version: 7.0.0]"
                    },
                    "locations": [{
                            "physicalLocation": {
                                "artifactLocation": {
                                    "uri": "file:///{C:/DynamoDSWorkspace}/{package}/bin/dll_assembly_name"
                                },
                                "region": {}
                            }
                        }
                    ]
                }
            ],
            "columnKind": "utf16CodeUnits"
        }
    ]
}
```
The code on L:79 is meant to get the 'real' dll assembly name which is located in (runs->results[] -> locations[] -> physicalLocation -> artifactLocation -> uri) in the AnalysisReport.sarif, replacing the 'file:///' by 'empty-string', the {workspace} path defined in appsetting.json plus '{package}/bin' by 'empty-string',  having as a result the '/dll_assembly_name' value

Forward slashes in (runs->results[] -> locations[] -> physicalLocation -> artifactLocation -> uri) does not affect the execution of DynamoPackagesAnalyzer neither in windows nor linux because we are only parsing the results for the csv file, not reading the file defined in uri